internal/controller/kyma_controller.go

```diff
@@ -405,21 +406,61 @@ func (r *KymaReconciler) handleDeletingState(ctx context.Context, kyma *v1beta2.
 		}
 
 		r.RemoteClientCache.Del(client.ObjectKeyFromObject(kyma))
-		if err := remote.RemoveFinalizerFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
+		if err := remote.RemoveFinalizersFromRemoteKyma(ctx, r.RemoteSyncNamespace); client.IgnoreNotFound(err) != nil {
 			err = fmt.Errorf("error while trying to remove finalizer from remote: %w", err)
 			r.enqueueWarningEvent(kyma, deletionError, err)
 			return ctrl.Result{}, err
 		}
 
-		logger.Info("removed remote finalizer")
+		logger.Info("removed remote finalizers")
+	}
+
+	relatedManifests, err := r.getRelatedManifestCRs(ctx, kyma)
```

Comment:
I think it makes sense you introduce a function named `cleanupManifestCRs` to encapsulate all this code, they are highly cohesion.